packages/fuels-core/src/types/wrappers/transaction.rs

```diff
@@ -75,7 +76,13 @@ pub trait Transaction: Into<FuelTransaction> + Send {
         parameters: &ConsensusParameters,
     ) -> Result<(), Error>;
 
-    fn id(&self, params: &ConsensusParameters) -> Bytes32;
+    fn precompute(&mut self, chain_id: u64) -> Result<(), Error>;
```

Comment:
I would make the `chain_id` type `ChainId`. Would save us from a lot of `.into()`s

```suggestion
    fn precompute(&mut self, chain_id: ChainId) -> Result<(), Error>;
```

Reply:
See comments about this.